app/controllers/subscriptions_controller.rb

```diff
@@ -30,15 +30,18 @@ def create
   def edit
     subscription = Subscription.find_and_verify_by_token(token)
     @subscription = SubscriptionPresenter.new(subscription)
+    @subscription_update_form = SubscriptionForm.new(subscription)
 
     Auditor::Audit.new(subscription, 'subscription.alert.edit', current_session_id).log_without_association
   end
 
   def update
     subscription = Subscription.find_and_verify_by_token(token)
     @subscription = SubscriptionPresenter.new(subscription)
+    @subscription_update_form = SubscriptionForm.new(subscription)
 
-    if @subscription.update(subscription_params)
+    if @subscription_update_form.valid?
```

Comment:
Need to add validations on the form for 
- Email presence, email
- Frequency presence
- At least one other field not blank